### DISCUSSION TITLE
USB MIDI

### DISCUSSION THREAD

**Discussion by mattia714** [marcel-licence/esp32_drum_computer#8] [Q&A]
It's possible to use the drum computer with USB MIDI using a PC as controller?
Maybe with software serial midi like hairless-midi?
Thanks in advance.

**Comment by marcel-licence** [comment]
Yes sure it is possible. That was also the way for me to generate and record some demo MIDI files.
In https://github.com/marcel-licence/esp32_drum_computer/blob/main/config.h the define MIDI_RECV_FROM_SERIAL configures the serial input to receive MIDI data. The baudrate is set by default to the 115200 (https://github.com/marcel-licence/esp32_drum_computer/blob/main/esp32_drum_computer.ino in Serial.begin(...))

**Reply by mattia714** [reply]
But in esp32_drum_computer.ino there are those lines
 Serial.begin(115200);
    Serial.println();
    Serial.printf("Loading data\n");
    Serial.printf("Firmware started successfully\n");

It seems to me that this is the output for the serial monitor

**Reply by marcel-licence** [reply]
Yes that is right. I didn't implement different modes. The output is just serial. The serial input can be interpreted as MIDI messages. You can ignore what is going out in this case. I didn't test hairless-midi yet. Do it have some trouble when getting invalid data?

**Comment by mattia714** [comment]
Hi Marcel.,

after some week I'm experimenting with my esp32 audio kit. 
I Upload the code without problem and I can hear the click.
Now I would like to modify z_config.ino to suite my needs.

I saw that every button or pot is defined with a string like this:
    { 0xD, 0x52, "stop", Sequencer_Stop, NULL, 0},
what is the meaning of the 6 values?

Thanks in advance,

Mattia
